### ISSUE TITLE
[HOLD for payment 2024-05-03] [$1000] Android - LHN - Unable to open reports, tasks and threads in LHN

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#40048]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.62-0
**Reproducible in staging?:** y
**Reproducible in production?:** n
**If this was caught during regression testing, add the test name, ID and link from TestRail:** n/a
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** Applause internal team
**Slack conversation:**

## Action Performed:
1. Launch the app and log in with an existing account with multiple chats
2. Open several reports, tasks, and threads

## Expected Result:
All reports, tasks, and threads can be opened

## Actual Result:
Some reports and tasks can be opened while I cannot open others. They are unresponsive to a tap.

## Workaround:
unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [ ] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

https://github.com/Expensify/App/assets/38435837/2be3e4f7-0bf5-4c09-a06d-23653b89e2d9

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~012a5ce21b6eda6732</li>
        <li>Upwork Job ID: 1778387468668194816</li>
        <li>Last Price Increase: 2024-04-17</li>
    </ul>
</details>

**Comment by m-natarajan** [comment]
We think this bug might be related to #vip-vsb

**Comment by Julesssss** [comment]
Can reproduce on Android, it seems that reports I haven't opened in a while won't open.

**Comment by dylanexpensify** [comment]
@Julesssss shall we make external?

**Comment by Julesssss** [comment]
@dylanexpensify I shared it in the channel seeking thoughts, but yeah lets open it up. I'll share it with contributors too and bump it to 500 for a solution that doesn't revert the architecture PR

**Comment by Beamanator** [comment]
FYI can't reproduce on iOS Native

**Comment by WoLewicki** [comment]
@ikevin127 we are using patch that brings the PR with new arch in `FlashList` to the App. We will investigate the issue further but I could not reproduce it locally so far.

**Comment by ikevin127** [comment]
@WoLewicki Thanks! I didn't know that we patched that PR. Then it must be something else breaking the `FlashList` 🧐

**Comment by ikevin127** [comment]
@WoLewicki Looks like the patch itself might've broken the FlashList based on https://github.com/Expensify/App/issues/40077#issuecomment-2050045059.

**Comment by WoLewicki** [comment]
Could you check if https://github.com/Expensify/App/pull/40191 solves the issue?

**Comment by dylanexpensify** [comment]
@parasharrajat would you mind giving us an update on where we're at and ETA?

**Comment by roryabraham** [comment]
I found a thread for which I could reproduce this issue on staging

**Comment by roryabraham** [comment]
I'm not sure if it's just really poor performance, but I'm unable to type any messages on Android dev on main

**Comment by roryabraham** [comment]
I'm blocked by testing this by https://github.com/Expensify/App/issues/40040

**Comment by ikevin127** [comment]
> Could you check if https://github.com/Expensify/App/pull/40191 solves the issue?

@WoLewicki Sorry for the delay! Just performed a build on Android: Native, latest main and here's my findings:
- **cannot** reproduce the open reports / task / threads side of this issue - it works as expected see video:

<details>
<summary>Video 1 - Android: Native (real device)</summary>

https://github.com/Expensify/App/assets/56457735/58565556-2737-49ce-805d-4558bfb1449b

</details>

- **instead**, opening threads is extremely slow from tap to open, sometimes requires multiple taps (see video 2)
- triggering context menu of a message by long-press is also really slow / janky and heavy on the UI (see video 2)

<details>
<summary>Video 2 - Android: Native (real device)</summary>

https://github.com/Expensify/App/assets/56457735/52923590-dd88-4623-99e2-04ddd9e357b7

</details>

^ - I'm assuming this is related to the new arch PR merged recently, not sure if anything specifically related to the **FlashList** patch since I tested with clean build of PR https://github.com/Expensify/App/pull/40191 and the the issues mentioned above seem to keep happening, here's the video:

<details>
<summary>Video 2 - Android: Native (real device)</summary>

https://github.com/Expensify/App/assets/56457735/3a9acc0e-a095-4076-948b-2fe84de8fb6c

</details>

**Summary:**
- as far as this issue goes, it looks like most of us can't reproduce the "open reports" side of it, except for specific reports
- looks like there are other issues related to opening tasks / threads and the context menu by long-press on a message

**Comment by roryabraham** [comment]
hoping this may be fixed by https://github.com/Expensify/App/pull/40191, will retest once it's deployed [here](https://github.com/Expensify/App/actions/runs/8713394572/job/23901702379)

**Comment by roryabraham** [comment]
Unfortunately I can confirm that this _is not_ fixed by https://github.com/Expensify/App/pull/40191 in 1.4.62-13

**Comment by kavimuru** [comment]
Issue is still reproducible in the build version 1.4.62-14 on pixel 6/ android 14

https://github.com/Expensify/App/assets/43996225/789986a5-8233-4478-b4c4-ce17e75241dd

[App_Info_1.4.62-14.json](https://github.com/Expensify/App/files/15011846/App_Info_1.4.62-14.json)

[Profile_trace_for_1.4.62-14.cpuprofile](https://github.com/Expensify/App/files/15011847/Profile_trace_for_1.4.62-14.cpuprofile)

**Comment by ikevin127** [comment]
Has anybody tried [my proposal](https://github.com/Expensify/App/issues/40048#issuecomment-2049738563) to see if it fixes the "Unable to open reports" part of this issue ?

[LHNOptionsList](https://github.com/Expensify/App/blob/b700bbee6035dc2492bd78629f9eda59228e97fb/src/components/LHNOptionsList/LHNOptionsList.tsx#L3) uses FlashList to render reports in LHN while `ReportActionsList` (report actions) use InvertedFlatList to render report threads and messages (context menu by long-press).

My only lead right now would be to replace `LHNOptionsList`'s FlashList with FlatList and perform an adhoc build for the testers to confirm if that fixes it, at least temporarily to unblock deploy.

On my side the issue is not reproducible...

cc @roryabraham @thienlnam

**Comment by thienlnam** [comment]
@ikevin127 Could you spin up a PR for it and then I can create an adhoc build for it - and we'll have someone that can reproduce try to see if they can do it on that branch

**Comment by ikevin127** [comment]
♻️ PR https://github.com/Expensify/App/pull/40380 ready for adhoc build and testing!

> [!note]
> 1. I used RN's FlatList instead of our own custom FlatList component.
> - **Why ?** Because when I simply replaced FlashList -> FlatList I had a weird scrolling issue that would keep resetting the scroll at the top when attempting to scroll down -> this was most likely happening since compared to RN's FlatList, our custom FlatList needs some extra crucial props passed to work as expected.
> 2. If the fix solves the issue I will make sure to actually use our custom FlatList component since that has some extra performance optimizations added.

cc @thienlnam

**Comment by roryabraham** [comment]
I'm pretty certain this has the same root cause as https://github.com/Expensify/App/issues/40108, which is more obvious to reproduce and where I left a number of breadcrumbs hopefully hinting at the root cause. So I'm going to throw this issue on HOLD for that one

**Comment by roryabraham** [comment]
taking this off HOLD because I was mistaken in thinking it would have the same root cause. I should've known because when this issue occurs, we _do_ see visual feedback for the touch event. So it seems like the touch is being correctly registered, which was not the case for https://github.com/Expensify/App/issues/40108.

Without further investigation, I'm more inclined to guess that this is a JS-level problem

**Comment by roryabraham** [comment]
Fortunately it's become more obvious how to reproduce this - just try to tap on any report in the LHN that is not in the initial frame of view. Scroll down just a bit and any row that wasn't fully in view before will not open

**Comment by roryabraham** [comment]
Removing the DeployBlockerCash label since this is Android-only and Android production deploys are temporarily disabled.

**Comment by roryabraham** [comment]
Update from @WoLewicki in slack:

> It seems like there are differences in how code is compiled on Android between debug and release configurations and due to this the contentOffset of ScrollView is not properly applied, resulting in buttons not being clickable after scrolling

**Comment by kbecciv** [comment]
@roryabraham Removing  the DeployBlockerCash label per your comment https://github.com/Expensify/App/issues/40048#issuecomment-2066868003

**Comment by roryabraham** [comment]
Unfortunately we don't have a substantive update or minimal reproduction here. Posted about this issue here: https://expensify.slack.com/archives/C06UTGZBWCA/p1713807467633649

**Comment by dylanexpensify** [comment]
Payment coming up

**Comment by dylanexpensify** [comment]
Payment coming up!

**Comment by thienlnam** [comment]
Was all internal / agency work - no payments needed here
